### PR TITLE
rust-clippy: fixes

### PR DESCRIPTION
**Pull request by petr-matousek**
this is a follow up to https://github.com/csutils/csmock/pull/157

### CODE REVIEW

**Comment by petr-matousek** [comment]
@lzaoral hi Lukas, this is a follow up to #157 based on your input

File: scripts/convert-clippy.py [hunk 1]
```diff
@@ -16,16 +16,17 @@ def main():
...
             continue
 
-        if (not (match := re.search(PACKAGE_ID_PATTERN, item["package_id"])) and
-            not (match := re.search(PACKAGE_ID_PATTERN2, item["package_id"]))):
-            continue
+        match = re.search(PACKAGE_ID_PATTERN, item["package_id"])
+        if not match:
+            match = re.search(PACKAGE_ID_PATTERN2, item["package_id"])
+            if not match:
+                continue
```
Comment (lzaoral) [inline scripts/convert-clippy.py:29]: note-to-self:  We should replace this with a `for` loop in the future when additional patterns will be needed to reduce nesting and code repetition.

**Review by lzaoral — CHANGES_REQUESTED** [review]
Thnak you, @petr-matousek   The clippy plugin now works on RHEL 8 as well!

Remarks:
* nit:  The commit message for c34fe3a looks incomplete. 🤔
* I have found out that if the conversion script fails, `csmock` will still exit with `0` which is quite unexpected.  Could you please fix that as well?  Thank you!

File: scripts/convert-clippy.py (hunk 1 above)
Comment (petr-matousek) [reply scripts/convert-clippy.py]: I tried to resolve your comments. please see the new set of commits.

File: make-srpm.sh [hunk 2]
```diff
@@ -95,7 +95,8 @@ BuildRequires: help2man
 %if 0%{?rhel} == 7
 %global python3_pkgversion 36
 
-# needed for /usr/share/csmock/scripts/convert-clippy.py
+# needed for /usr/share/csmock/scripts/convert-clippy.py because byte-compilation on el7 was
+# failing as el7 rpmbuild defaults to Python 2.7 for Python scripts in unversioned directories
```
[the remark this answers is not in the document]
Comment (petr-matousek) [reply make-srpm.sh]: I tried to resolve your comments. please see the new set of commits.

File: scripts/convert-clippy.py (hunk 1 above)
Comment (lzaoral) [reply scripts/convert-clippy.py]: Thank you very much, resolving this particular comment was not necessary!

File: scripts/convert-clippy.py [hunk 3]
```diff
@@ -10,21 +10,24 @@
...
 def main():
     for line in sys.stdin:
         try:
             item = json.loads(line)
         except Exception as e:
-            sys.stderr.write(f"rust-clippy: Error while converting results: {e}\n")
-            sys.stderr.flush()
-            continue
+            print("rust-clippy: Error while converting results:", e, file=sys.stderr)
+            exit(1)
```
[the remark this answers is not in the document]
Comment (lzaoral) [reply scripts/convert-clippy.py]: @petr-matousek See ^.

[the remark this answers is not in the document]
Comment (lzaoral) [reply scripts/convert-clippy.py]: The main issue with the exit code being ignored is still present and is cause by this piece of code:
https://github.com/csutils/csmock/blob/6b088e36d38a32b99b264b056c06e53e2fbe15f5/py/plugins/clippy.py#L41-L42

You have to enable the `pipefail` BASH shell option:
```shell
$ bash -c 'false | :'; echo $?
0
$ bash -c 'set -o pipefail; false | :'; echo $?
1
```

[the remark this answers is not in the document]
Comment (petr-matousek) [reply scripts/convert-clippy.py]: my shell programming abilities are obviously lacking behind. thanks for the hint -)

[the remark this answers is not in the document]
Comment (lzaoral) [reply scripts/convert-clippy.py]: Thank you for the fixup!

**Review by lzaoral — APPROVED** [review]
LGTM!  Thank you @petr-matousek!

**Comment by kdudka** [comment]
Merging, thank you both!

**Pull request merged**